lib/getCollectors.tsx

```diff
@@ -0,0 +1,30 @@
+const getCollectors = (data) => {
+  const aggregated = {}
+  const usersSet = new Set()
+
+  data.forEach((item) => {
+    const tokenId = item.args.tokenId.toString() // Convert BigInt to string for use as a key
+    const user = item.args.user
+
+    // If the tokenId doesn't exist in the aggregated object, create it
+    if (!aggregated[tokenId]) {
+      aggregated[tokenId] = {
+        address: item.address, // Assuming address is the same for each tokenId
+        permissions: item.args.permissions,
+        tokenId: tokenId,
+        users: [],
+      }
+    }
+
+    // Push the user into the array, avoiding duplicates
```

Comment:
please remove all dev comments in this file.
- code should be self-documenting.